src/main/java/Main.java

```diff
@@ -50,56 +58,77 @@ private static BattleshipModel getModelFromReq(Request req){
     //}
     //For testing purposes
 
-    private static Boolean placeShip(Request req, Response res) {
+    private static String placeShip(Request req) {
         //collect info to put this ship into the model
         BattleshipModel model = game.getModel();
         //get the name
-        String name = req.attribute("name");
+        String name = req.params("id");
+
         //get the start
-        Coordinate start = new Coordinate(req.attribute("Across"), req.attribute("Down"));
+        Coordinate start = new Coordinate(Integer.parseInt(req.params("col")), Integer.parseInt(req.params("row")));
+
         //get orientation
-        String orientation = req.attribute("orientation");
+        String orientation = req.params("orientation");
+
         //set the scan boolean variable
-        boolean canScan = (name.equals("submarine"));
+        boolean canScan = true;
+
         //get length
-        int length = req.attribute("size");
+        int length=0;
+        if(name.equals("aircraftCarrier")){
+            length=5;
+        }
+        if(name.equals("clipper")){
+            length=3;
+        }
+        if(name.equals("submarine")){
+            canScan=false;
+            length=3;
+        }
+        if(name.equals("dinghy")){
+            length=1;
+        }
+        if(name.equals("battleship")) {
+            length = 4;
+        }
         //get end
         Coordinate end;
         if(orientation.equals("horizontal")){
-            end = new Coordinate(start.getAcross()+length, start.getDown());
+            end = new Coordinate(start.getAcross()+length-1, start.getDown());
         }
         else{
-            end =  new Coordinate(start.getAcross(), start.getDown()+length);
+            end =  new Coordinate(start.getAcross(), start.getDown()+length-1);
         }
+        System.out.println("It is the place ship's fault");
```

Comment:
Left a print statement in, I happen to know it was for testing, but just a heads up.